### PR TITLE
#2106 - Fix reports not getting deleted when soft deleting a post

### PR DESCRIPTION
**Pull request by Naththo**
#2106

### CODE REVIEW

**Comment by JN-Jones** [comment]
The issue should be fixed in class_moderation.php. Also you don't delete reports for the firstpost if the thread is deleted.

**Comment by Destroy666x** [comment]
Now it only marks reports for threads, what about posts?

File: moderation.php [hunk 1]
```diff
@@ -2739,8 +2738,9 @@
 		}
 
 		$moderation->soft_delete_posts($pids);
-
 		log_moderator_action($modlogdata, $lang->multi_soft_delete_posts);
+
```
Comment (Destroy666x) [inline moderation.php]: Please delete one extra line

**Comment by JN-Jones** [comment]
Also note that there isn't a report type named "threads", the first post is used instead. Also if a thread is soft deleted all posts should be checked imho.

**Comment by Destroy666x** [comment]
@JN-Jones, not sure what you mean,threads is correct: https://github.com/mybb/mybb/blob/feature/inc/functions.php#L4574

**Comment by Destroy666x** [comment]
Yep, I can confirm it works fine with both single and inline thread/post soft deletion. Reports for all posts in a soft deleted threads are marked.

**Comment by JN-Jones** [comment]
Ah, thought `mark_reports` only accepts the types we store in the database - yeah, it'll mark all posts with that query.

**Pull request merged**